### PR TITLE
🏗️ Maintenance week 47: minor fixes, cleanup and new workflow with layered requirements 

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

- ADDS: Completes list of services in e2e logs dump
- FIXES setup in director-v2 unit-test in travis
- ADDS: all services have startup banner (except for api-server, which is done in another branch) 
- Missing issues from previous review from @GitHK
- 🎉 NEW workflow with [layered requirements](https://github.com/jazzband/pip-tools#workflow-for-layered-requirements).   
       -  Every ``requirement/_*.txt`` includes ONLY dependencies from that layer. Then, they are stacked to achieve different configurations in ``requirement/dev|ci|prod.txt``.
       - This is achieved by including constraints (i.e. ``-c``) instead of requirements (i.e. ``-r``). The latter installs the included dependencies while the former only considers those constraints to resolve version dependencies. 
- 🎉 NEW setup [pre-commit](https://pre-commit.com/) hooks to run git pre-commit hooks. Activated checks as auto-format, check on large files, ... on committed files 
- FIXES **e2e**: Replaced uuids starting with ``template-uuid-...`` from fake projects since it does not pass anymore the jsonschema validation

### Dependencies

- [yarl 1.6.3](https://github.com/aio-libs/yarl/releases/tag/v1.6.3) patch upgrade
- [aiohttp 3.7.3](https://github.com/aio-libs/aiohttp/releases/tag/v3.7.3) patch upgrade
- [requests 2.25](https://github.com/psf/requests/compare/v2.24.0...v2.25.0) minor upgrade
- [dataclasses 0.8](https://github.com/ericvsmith/dataclasses/commit/95d2a4ca30e9dfdbc6837921cc5a56b89364be62) which is the same as 0.7 (just doc changes)
- update **ONLY dependencies used for testing** (e.g. tooling or fixtures)  

## How to test

Setup standard devenv workflow on a services: create environ (``make devenv``), cd to services (``cd services/{}), install it (``make install-dev``) and run tests (``make test-dev-unit``)

## Checklist

- [x] Did you change any service's API? Then make sure to bundle document and upgrade version (``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [x] Unit tests for the changes exist
- [x] Runs in the swarm
- [x] Documentation reflects the changes
- [x] New module? Add your github username to [.github/CODEOWNERS](.github/CODEOWNERS)

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      - id: end-of-file-fixer
+      - id: check-yaml
+      - id: check-added-large-files
+      - id: check-merge-conflict
+  - repo: https://github.com/PyCQA/isort
+    rev: 5.6.4
+    hooks:
+      - id: isort
+  - repo: https://github.com/psf/black
+    rev: 20.8b1
```
Comment (sanderegg) [inline .pre-commit-config.yaml:20]: :+1:

File: ci/travis/unit-testing/director-v2.bash [hunk 2]
```diff
@@ -34,7 +34,7 @@ script() {
     then
         pytest --numprocesses=auto --cov=simcore_service_director-v2 --durations=10 --cov-append \
           --color=yes --cov-report=term-missing --cov-report=xml --cov-config=.coveragerc \
-          -v -m "not travis" services/director-v2/tests
+          -v -m "not travis" services/director-v2/tests/unit
```
Comment (sanderegg) [inline ci/travis/unit-testing/director-v2.bash:37]: good catch!

File: requirements/devenv.in [hunk 3]
```diff
@@ -13,12 +13,14 @@
...
 # dependency manager
 pip-tools
+
 # version manager
 bump2version
+
 # renaming
 rope
+
+# git hooks. pre-commit fails to install directly!
```
Comment (sanderegg) [inline requirements/devenv.in:25]: this is a bit of a strange comment. not sure I understand what that really means?

**Review by sanderegg — APPROVED** [review]

File: requirements/devenv.in (hunk 3 above)
Comment (pcrespov) [reply requirements/devenv.in:25]: True, I do not fully understand it myself right now. I will rewrite it.
The thing is as follows: when you try to ``pip install pre-commit``in a fresh venv it warns about incompatible constraints. My guess is that the problem is in the pre-commit's requirements setup. In order to solve the dependencies correctly we use here ``pip-compile``

**Pull request merged**